R/utilities.R

```diff
@@ -2731,6 +2906,50 @@ collate_lymphgen = function(sample_table,
 }
 
 
+#' Expand a sample_table (meta data) horizontally with different flavours of lymphgen data.
+#'
+#' @param sample_table Input data frame with sample IDs in the firs column.
+#'
+#' @return
+#' @export
+#' @import tidyverse
+#'
+#' @examples
+#' metadata = get_gambl_metadata()
+#' colalted_meta = collate_lymphgen(sample_table = metadata)
+#' 
+collate_lymphgen = function(sample_table){
```

Comment:
Also it would be better to call this argument `these_samples_metadata` for consistency with a lot of other GAMBLR functions. It could also be optional to join it to the metadata table, and could alternatively be provided with a list of sample_ids via `these_sample_ids`.

Reply:
What if we made these_samples_metadata optional and had the function return the rows for all samples in the files by default?

Reply:
Agree, Ryan,  this would also be consistent with a lot of other GAMBLR functions.

Reply:
Sort of. Many GAMBLR functions default to calling get_gambl_metadata under the hood. In this scenario, I'm saying the function instead rely on the sample IDs in the files it's loading (ignoring the metadata entirely)

Reply:
`these_samples_metadata` has been added as an optional parameter (replacing sample_table) if provided, lymphgen will be joined with this data frame. If not provided, the function will return the rows for all the samples in the files (default).